src/components/cards/RelationshipsCard.jsx

```diff
@@ -1,106 +1,510 @@
+      messageId="CONFIRM_REMOVE_RELATIONSHIP"
+    />,
+    <StandardDialog
+      open={openRelationshipDialog}
+      onClose={onCloseDialog}
+      titleId={'ADD_RELATIONSHIP'}
+    >
+      <DialogContent>
+        <IndividualSelector
+          setSelectedIndividualId={setSelectedIndividualId}
+          setSelectedIndividualFirstName={
+            setSelectedIndividualFirstName
+          }
+        />
+        <div
+          id="rel-type"
+          style={{ width: 'fit-content', minWidth: 470 }}
+        >
+          <Autocomplete
+            id="types"
+            clearOnEscape
+            style={{ marginTop: 12 }}
+            value={currentType}
+            options={types}
+            renderOption={option => (
+              <Text value={option.guid}>{option.label}</Text>
+            )}
+            onChange={(_, newValue) => onChangeType(newValue)}
+            getOptionLabel={option => get(option, 'label', '')}
+            renderInput={params => {
+              return (
+                <TextField
+                  {...params}
+                  style={{ width: 280 }}
+                  variant="standard"
+                  label={intl.formatMessage({
+                    id: 'SELECT_RELATIONSHIP_TYPE',
+                  })}
+                />
+              );
+            }}
+          />
+        </div>
+        {currentRoles?.length > 0 && [
+          <div
+            id="roles1-div"
+            style={{ width: 'fit-content', minWidth: 470 }}
+          >
+            <Autocomplete
+              id="roles1"
+              clearOnEscape
+              style={{ marginTop: 12, width: 'fit-content' }}
+              value={currentRole2}
+              options={currentRoles}
+              renderOption={option => (
+                <Text value={option.guid}>{option.label}</Text>
+              )}
+              onChange={(_, newValue) => onChangeRole2(newValue)}
+              getOptionLabel={option => get(option, 'label', '')}
+              getOptionSelected={(option, val) =>
+                option.guid ? option.guid === val : false
+              }
+              renderInput={params => {
+                return (
+                  <TextField
+                    {...params}
+                    style={{ width: 280 }}
+                    variant="standard"
+                    label={intl.formatMessage(
+                      { id: 'SELECT_RELATIONSHIP_ROLE_1' },
+                      {
+                        ind1: individualFirstName
+                          ? individualFirstName
+                          : intl.formatMessage({
+                              id: 'UNNAMED_INDIVIDUAL',
+                            }),
+                      },
+                    )}
+                  />
+                );
+              }}
+            />
+          </div>,
+          <div
+            id="roles2-div"
+            style={{ width: 'fit-content', minWidth: 470 }}
+          >
+            <Autocomplete
```

Comment:
feel like a lot of these Autocomplete props are the same as the two right above. you could create a component in another file to minimize the complexity of this file and DRY up the code a bit

Reply:
Done